### DISCUSSION TITLE
The number of edges are reduced in convertion of heterogeneous graph to networkx

### DISCUSSION THREAD

**Discussion by Z-Rajaei** [pyg-team/pytorch_geometric#8547] [Q&A]
Hello,
I have a heterogeneous graph. I used the following code to convert it to a networkx graph:
g = torch_geometric.utils.to_networkx(data.to_homogeneous(), to_undirected=False)

The number edges in the original graph is 297. when it is converted to homogeneous graph, it is also 297. But when it is converted to networkx, it will becom 293!! why??

homogeneous graph is: Data(edge_index=[2, 297], batch=[97], node_type=[97], edge_type=[297])
networkx graph is: DiGraph with 97 nodes and 293 edges

**Comment by rusty1s** [comment]
I am not totally sure, but `networkx` conversion may drop duplicated edges.

**Reply by Z-Rajaei** [reply]
You're right.
Is there any way to convert the graph to MultiDiGraph to maintain the duplicated edges? because the edge names are different.

**Reply by rusty1s** [reply]
Added this here: https://github.com/pyg-team/pytorch_geometric/pull/8575
